frontend/src/general/DeleteDeviceButton.jsx

```diff
@@ -23,14 +40,15 @@ export default function DeleteDeviceButton(props) {
   const confirmDelete = () => {
     DeviceApi.deleteDevice(deleteId);
     history.push("/Devices");
+    history.go(0);
     return setOpen(false);
   };
 
   return (
     <>
-      <Button id="deleteBtn" variant="outlined" onClick={openDeleteDialog}>
-        Delete
-      </Button>
+      {button
+        ? renderDeleteButton(openDeleteDialog)
+        : renderDeleteMenuItem(openDeleteDialog)}
```

Comment:
I really like how you did this to avoid code duplication by applying conditional rendering!